report_substitute/models/ir_actions_report.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class IrActionReport(models.Model):
+
+    _inherit = 'ir.actions.report'
+
+    action_report_substitution_rule_ids = fields.One2many(
+        comodel_name="ir.actions.report.substitution.rule",
+        inverse_name="action_report_id",
+        string="Substitution Rules",
+    )
+
+    @api.multi
+    def _get_substitution_report(self, model, active_ids):
```

Comment:
Can you add a limit=1, increase every time and once it has reached 50 raise a Exception?
This way at least we know if we have recursion issue. What do you think?
It is not blocking.

Reply:
I added a check for the substitution infinite loop when creating the rules. can you check it please.